### PR TITLE
fix #275552 fix#137501: Inability to show multi-measure rests in Continuous View

### PR DESCRIPTION
**Pull request by handrok**
1) The reason of crash is utilizing already deleted pointer. Each element are lying in the tree.   The parent of VBox element is System element. When we change mode to continuous view MS deleted all System elements and create new one. Then populate new tree with new System which is root. But VBox is not shown in continuous view and MS do not set new parent for VBox. Sow VBox have pointer to deleted Sytem element. I fixed it by setting parent of Vbox nullptr.

2) Add header files to libmscore/CMakeLists. It doesn't effect on building solution. But it is necessary to include header files to Visual Studio IDE

### CODE REVIEW

File: libmscore/element.cpp [hunk 1]
```diff
@@ -375,12 +375,12 @@ QPointF Element::pagePos() const
...
       QPointF p(pos());
-      if (parent() == 0)
+      if (parent() == nullptr)
             return p;
 
       if (_flags & ElementFlag::ON_STAFF) {
-            System* system = 0;
-            Measure* measure = 0;
+            System* system = nullptr;
+            Measure* measure = nullptr;
```
Comment (jthistle) [inline libmscore/element.cpp]: I might be wrong, but isn't the default value of an uninitialised pointer `nullptr`?
So you could just write `System* system;`, for example?

Comment (MarcSabatella) [reply libmscore/element.cpp]: No.  Local variables need to be initialized.  Only globals / static variables get a predictable default value.

Comment (jthistle) [reply libmscore/element.cpp]: Ah thanks, sorry about that.

**Comment by handrok** [comment]
The another reason of crash is founded. To fix it I fix the following issue https://musescore.org/en/node/137501 "inability to show multi-measure rests in Continuous View "

**Comment by handrok** [comment]
Refactoring is needed. See following methods: 
void LayoutContext::layoutLinear()
System* Score::collectSystem(LayoutContext& lc)
Code duplication is presence.

**Comment by mattmcclinch** [comment]
Nice job getting getting this to work! I can still trigger a crash by toggling multi-measure rests in Continuous View if I try hard enough, and as it turns out, the crash happens in [code that I committed](https://github.com/musescore/MuseScore/blob/7354ddc0eea621d94eb9c3368e7b135929c22e27/libmscore/clef.cpp#L151). As far as I can tell, the reason is because the measure's parent is not set to _nullptr_ when the system is deleted. [This small change](https://github.com/mattmcclinch/MuseScore/commit/3d87d09a20d4a64c4f951fa95aa365e9b5ae93ce) prevents the crash from happening.

**Comment by handrok** [comment]
@mattmcclinch thanks. Add your changes to PR

**Comment by mattmcclinch** [comment]
You will probably want to add [this check](https://github.com/mattmcclinch/MuseScore/commit/5e18f8c77196e908e40edc23d1bdebc289e45642) in order to diagnose [issue 267726](https://musescore.org/en/node/267726) on this branch. Otherwise, when the TBox is encountered, _lc.curMeasure_ will never advance, resulting in an infinite loop. Of course, that issue still remains, but this will make it easier to debug.

**Pull request merged**